### PR TITLE
IGNITE-10371: [ML] Add multiple metrics calculation fo Binary Classification Evaluation process

### PR DESCRIPTION
**Pull request by zaleslaw**

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    /** Returns the pair of metric name and metric value. */
+    public Map<String, Double> toMap() {
+        Map<String, Double> metricValues = new HashMap<>();
+        for (Field field : this.getClass().getDeclaredFields())
+            try {
+                metricValues.put(field.getName(), field.getDouble(this));
+            } catch (IllegalAccessException e) {
+                e.printStackTrace();
```
Comment (avplatonov) [inline modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java]: I don't think that exception hiding is valid behavior for this class

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    private double tn;
+
+    /** False Positive (FP). */
+    private double fp;
+
+    /** False Negative (FN). */
+    private double fn;
+
+    /** Sensitivity or True Positive Rate (TPR). */
+    private double recall;
```
Comment (avplatonov) [inline modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java]: I thought that you just add tp,tn,fp and fn values and calculate other metrics just-in-time by methods like accuracy() 
I realize that it counter-productive but computing of such metrics is a cheap procedure. IMHO a lot of fields are non-readable especially with setters.

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        this.fn = fn;
+    }
+
+    /** */
+    public double recall() {
+        return recall;
+    }
+
+    /** */
+    public void setRecall(double recall) {
```
Comment (avplatonov) [inline modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java]: Moreover setter for field dependent on other fields can break state of class

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java (hunk 2 above)
Comment (zaleslaw) [reply modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java]: Good point, will think about that

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java (hunk 1 above)
Comment (zaleslaw) [reply modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java]: Should we rethrow something IgniteML related do yu think?

**Pull request closed without merging**